R/input-date.R

```diff
@@ -124,19 +125,103 @@ dateInput <- function(inputId, label, value = NULL, min = NULL, max = NULL,
                `data-date-days-of-week-disabled` =
                    jsonlite::toJSON(daysofweekdisabled, null = 'null')
     ),
-    datePickerDependency
+    datePickerDependencies(sassVariables)
   )
 }
 
-datePickerDependency <- htmlDependency(
-  "bootstrap-datepicker", "1.6.4", c(href = "shared/datepicker"),
-  script = "js/bootstrap-datepicker.min.js",
-  stylesheet = "css/bootstrap-datepicker3.min.css",
-  # Need to enable noConflict mode. See #1346.
-  head = "<script>
-(function() {
-  var datepicker = $.fn.datepicker.noConflict();
-  $.fn.bsDatepicker = datepicker;
-})();
-</script>"
-)
+datePickerDependencies <- function(sassVariables = list()) {
+
+  if (useBsTheme() || length(sassVariables) == 0) {
+    scssDir <- system.file(package = "shiny", "www", "shared", "datepicker", "scss")
+    tmpDir <- tempfile("datepicker-scss")
+    dir.create(tmpDir)
+    outFile <- file.path(tmpDir, "custom.css")
+    sassFunc <- if (useBsTheme()) bootstraplib::bootstrap_sass else sass::sass
+    defaults <- list(
+      sassVariables %OR% "",
+      bsThemeDateDefaults()
+    )
+    sassFunc(
+      list(
+        defaults,
+        sass::sass_file(file.path(scssDir, "build3.scss"))
+      ),
+      output = outFile
+    )
+    # Hash the skin and defaults so that the html dependency from one date-picker won't stomp another
+    cssDependency <- htmlDependency(
+      paste0("datepicker-datepicker-css-", digest::digest(defaults, "xxhash64")),
+      version = "1.9.0",
+      src = dirname(outFile),
+      stylesheet = basename(outFile)
+    )
+  } else {
+    cssDependency <- htmlDependency(
+      "bootstrap-datepicker-css",
+      version = "1.9.0",
+      src = c(href = "shared/datepicker"),
+      stylesheet = "css/bootstrap-datepicker3.min.css",
+    )
+  }
+
+  list(
+    cssDependency,
+    htmlDependency(
+      "bootstrap-datepicker-js",
+      version = "1.9.0",
+      c(href = "shared/datepicker"),
+      script = "js/bootstrap-datepicker.min.js",
+      # Need to enable noConflict mode. See #1346.
+      head = "<script>(function() {
+        var datepicker = $.fn.datepicker.noConflict();
+        $.fn.bsDatepicker = datepicker;
+      })();
+     </script>"
+    )
+  )
+}
+
+
+
+bsThemeDateDefaults <- function() {
+  if (!useBsTheme()) {
+    return("")
```

Comment:
Would a `list()` make more sense, to be consistent with the other possible return values?